contracts/managers/triggers/RSITrendingTrigger.sol

```diff
@@ -0,0 +1,186 @@
+ * the RSI Trending trigger is bullish.
+ *
+ * This trigger is for trend trading strategies which sets upperBound as resistance
+ * and lowerBound as support. When RSI level crosses above upperBound the indicator
+ * is bullish. When RSI level crosses below lowerBound the indicator is bearish.
+ *
+ */
+contract RSITrendingTrigger is
+    ITrigger
+{
+    using SafeMath for uint256;
+
+    /* ============ State Variables ============ */
+    IMetaOracleV2 public rsiOracleInstance;
+    // RSI Bound under which strategy indicates bearish market
+    uint256 public lowerBound;
+    // RSI Bound over which strategy indicates bullish market
+    uint256 public upperBound;
+    uint256 public rsiTimePeriod;
+    uint256 public triggerFlippedIndex;
+    bool private lastConfirmedTrendState;
+    
+    bool public requiresConfirmation = false;
+
+    /*
+     * RSITrendingTrigger constructor.
+     *
+     * @param  _rsiOracleInstance       The address of RSI oracle
+     * @param  _lowerBound              Lower bound of RSI to trigger a rebalance
+     * @param  _upperBound              Upper bound of RSI to trigger a rebalance
+     * @param  _rsiTimePeriod           The amount of days to use in RSI calculation
+     * @param  _initialTrendState       Starting state based on current trend
+     */
+    constructor(
+        IMetaOracleV2 _rsiOracleInstance,
+        uint256 _lowerBound,
+        uint256 _upperBound,
+        uint256 _rsiTimePeriod,
+        bool _initialTrendState
+    )
+        public
+    {
+        // Check that upper bound value must be greater than lower bound value
+        require(
+            _upperBound >= _lowerBound,
+            "RSITrendingTrigger.constructor: Upper bound must be greater than lower bound"
+        );
+
+        // Set all state variables
+        rsiOracleInstance = _rsiOracleInstance;
+        lowerBound = _lowerBound;
+        upperBound = _upperBound;
+        rsiTimePeriod = _rsiTimePeriod;
+        lastConfirmedTrendState = _initialTrendState;
+        triggerFlippedIndex = 0;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Since RSI does not require a confirmation leave initialTrigger function unimplemented.
+     */
+    function initialTrigger()
+        external
+    {}
+
+    /*
+     * If RSI is above upper bound then should be true, if RSI is below lower bound
+     * then should be false. If in between bounds then returns the state of the current trend.
+     */
+    function confirmTrigger()
+        external
+    {
+        // Query RSI oracle
+        uint256 rsiValue = rsiOracleInstance.read(rsiTimePeriod);
+
+        // Check RSI trend is different from last confirmed trend state 
+        require(
+            isNewRSIState(rsiValue),
+            "RSITrendingTrigger.confirmTrigger: Current RSI value does not change trend state."
+        );
+
+        lastConfirmedTrendState = rsiValue >= upperBound ? true : false;
```

Comment:
Potentially put this in a private function - since this logic is duped in Line 178. If the intention is for them to be the same, it could result in less bugs /errors if changed.